server/crashmanager/forms.py

```diff
@@ -0,0 +1,131 @@
+            Field('component', wrapper_class='col-md-6'),
+        ),
+        Row(
+            Field('whiteboard', wrapper_class='col-md-6'),
+            Field('keywords', wrapper_class='col-md-6'),
+        ),
+        Row(
+            Field('op_sys', wrapper_class='col-md-6'),
+            Field('platform', wrapper_class='col-md-6'),
+        ),
+        Row(
+            Field('cc', wrapper_class='col-md-6'),
+            Field('assigned_to', wrapper_class='col-md-6'),
+        ),
+        Row(
+            Field('priority', wrapper_class='col-md-6'),
+            Field('severity', wrapper_class='col-md-6'),
+        ),
+        Row(
+            Field('alias', wrapper_class='col-md-6'),
+            Field('qa_contact', wrapper_class='col-md-6'),
+        ),
+        Row(
+            Field('version', wrapper_class='col-md-6'),
+            Field('target_milestone', wrapper_class='col-md-6'),
+        ),
+        'attrs',
+        'description',
+        'security',
+        Row(Field('security_group', wrapper_class='col-md-6')),
+        Row(Field('testcase_filename', wrapper_class='col-md-6')),
+        Submit('submit', 'Save', css_class='btn btn-danger'),
+        HTML("""<a href="{% url 'crashmanager:templates' %}" class="btn btn-default">Cancel</a>""")
+    )
+
+    class Meta:
+        model = BugzillaTemplate
+        fields = [
+            'name',
+            'summary',
+            'product',
+            'component',
+            'whiteboard',
+            'keywords',
+            'op_sys',
+            'platform',
+            'cc',
+            'assigned_to',
+            'priority',
+            'severity',
+            'alias',
+            'qa_contact',
+            'version',
+            'target_milestone',
+            'attrs',
+            'description',
+            'security',
+            'security_group',
+            'testcase_filename',
+        ]
+
+        labels = {
+            'name': 'Template name',
+            'summary': 'Summary',
+            'product': 'Product',
+            'component': 'Component',
+            'whiteboard': 'Whiteboard',
+            'keywords': 'Keywords',
+            'op_sys"': 'OS',
+            'platform': 'Platform',
+            'cc': 'Cc',
+            'assigned_to': 'Assigned to',
+            'priority': 'Priority',
+            'severity': 'Severity',
+            'alias': 'Alias',
+            'qa_contact': 'QA',
+            'version': 'Version',
+            'target_milestone': 'Target milestone',
+            'attrs': 'Custom fields',
+            'description': 'Bug description',
+            'security': 'This is a security bug',
+            'security_group': 'Security group',
+            'testcase_filename': 'Filename that will be used for the testcase',
+        }
+
+        widgets = {}
+        for field in fields:
+            if field not in ['description', 'attrs', 'security']:
+                widgets[field] = Textarea(attrs={'rows': 1})
```

Comment:
Just curious but is there a reason you're using textarea for all of these inputs?  It makes sense for `description` and possibly `attrs` but shouldn't the rest just be inputs?

Reply:
In `BugzillaTemplate` class all fields are declared as `models.TextField()` ones.

Django automatically converts them to `TextAreas` when using forms (see: [https://docs.djangoproject.com/en/3.1/ref/models/fields/#textfield](https://docs.djangoproject.com/en/3.1/ref/models/fields/#textfield)) so I kept this "convention".

But if you want, I could easily change those to display `TextInputs` instead.

Reply:
That was sloppy on our part in defining the template model... I think they probably should be `CharField(max_length=4095)`? If you'd rather specify `TextInput` instead, that's fine.